alerta/utils/mailer.py

```diff
@@ -55,7 +55,7 @@ def send_email(self, email, subject, body):
                 mx.set_debuglevel(True)
             mx.ehlo()
 
-            if self.smtp_starttls:
+            if not self.smtp_use_ssl and self.smtp_starttls:
                 mx.starttls()
```

Comment:
It seems like a good idea.

Or maybe we should use something like Flask-Mail ? What do you think ?

Reply:
Looks like Flask-Mail handles SSL and TLS the same way as Alerta currently does ...

**Flask-Mail**
```python
    def configure_host(self):
        if self.mail.use_ssl:
            host = smtplib.SMTP_SSL(self.mail.server, self.mail.port)
        else:
            host = smtplib.SMTP(self.mail.server, self.mail.port)


        host.set_debuglevel(int(self.mail.debug))


        if self.mail.use_tls:
            host.starttls()
        if self.mail.username and self.mail.password:
            host.login(self.mail.username, self.mail.password)


        return host
```

**Alerta Mail**
https://github.com/alerta/alerta/blob/4af78277e88d407fc30d6e6f0e4b03b3d363646f/alerta/utils/mailer.py#L48-L59

But Flask-Mail doesn't seem to support SSL `keyfile`, `certfile` and the code-base was last updated 4 years ago. Someone recently submitted a PR to add support for it but nothing has happened. See https://github.com/mattupstate/flask-mail/pull/162

Reply:
I saw that. But I never add any issue with Flask-Mail. My mail server use a dedicated port with tls (465). But with Alerta, if I do not set MAIL_STARTTLS=False, the connection doesn't work. I think that's because the default is False in Flask-Mail. So we should change the default in Alerta or move starttls() like you asked. What do you think is best ?
About the client certificate, keyfile and certfile are deprecated and a context should be used instead : https://docs.python.org/3.5/library/smtplib.html#smtplib.SMTP_SSL .